operators/src/engine/query_processor.rs

```diff
@@ -231,3 +231,25 @@ impl TypedVectorQueryProcessor {
         }
     }
 }
+
+/// Maps a `TypedVectorQueryProcessor` to another `TypedVectorQueryProcessor` by calling a function on its variant.
+/// Call via `map_typed_vector_query_processor!(input, processor => function)`.
+#[allow(unused_macros)]
```

Comment:
why not macro_export instead?

Reply:
`macro_export` puts the function to the root of the crate, is this intended?

Reply:
I think it's okay to put them at the root and also use them in the service crate.

Reply:
302bab9